src/services/stac/index.tsx

```diff
@@ -16,21 +18,30 @@ import { StacCollectionsResponse } from './types';
 // 2. Or, retrieve a limited number of collections and paginate on the server side
 // The latter requires more network requests, however it's more efficient for large datasets
 export const getStacCollections = async (
+  privateCatalog: string,
   searchQuery: string = '',
   limit: number = 99999,
 ): Promise<Collection[]> => {
-  const url = `${import.meta.env.VITE_STAC_ENDPOINT}/collections?limit=${limit}&q=${searchQuery}`;
+  const url = privateCatalog
+    ? `${import.meta.env.VITE_STAC_ENDPOINT}/catalogs/${privateCatalog}/collections?limit=${limit}&q=${searchQuery}`
+    : `${import.meta.env.VITE_STAC_ENDPOINT}/collections?limit=${limit}&q=${searchQuery}`;
+
   try {
     const response = await fetch(url, {
       method: 'GET',
+      credentials: 'include',
```

Comment:
Is there a better way that we can handle this? i.e. checking if we are running on localhost?

Reply:
I don't know, I'm not a fan of adding mode checks as they are easy to miss, or get wrong, but if we were to go down that road, I guess it would be something like:

```JavaScript
...(import.meta.env.PROD && {credentials : 'include'})
```

I've not proven the above works, but I don't see why it wouldn't. I would probably prefer that APIs in `DEV` environment accept requests from `localhost`, than this though, I personally find that cleaner e.g. in Django `settings/development.py`

```Python
ALLOWED_HOSTS = ["*"]
```
The CORS error is caused by the server, not the client, so I feel it is a server issue, not a client one, but I'm sure others may disagree with that opinion.

Reply:
^^
This refers to the CORS that comes through the OPA. I tried to resolve it a while ago and it seems unresolvable.